libraries/apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/codegen/kotlin/file/EnumAsEnumBuilder.kt

```diff
@@ -63,7 +64,9 @@ internal class EnumAsEnumBuilder(
           values.forEach { value ->
             addEnumConstant(layout.enumAsEnumValueName(value.targetName), value.enumConstTypeSpec())
           }
-          addEnumConstant("UNKNOWN__", unknownValueTypeSpec())
+          if (withUnknown) {
+            addEnumConstant("UNKNOWN__", unknownValueTypeSpec())
+          }
```

Comment:
Fun fact, server types cannot reuse the client types because of this. While a client might receive an unknown enum, it is never the case for servers.